### PR TITLE
fix: Update pre-populated programme membership details

### PR DESCRIPTION
**Pull request by babul3012**
* Using most recent ProgrammeMembership to pre-populate in FormRPartA
* Using the MEDICAL_CURRICULUM for programme specialty
* Moved drop down options to create page

TISNEW-4241

### CODE REVIEW

File: src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
   const pd = traineeProfile.personalDetails;
-  const programme = traineeProfile.programmeMemberships[0];
+  const programme = traineeProfile.programmeMemberships
+    .filter(pm => new Date(pm.startDate) < new Date())
+    .reduce(function(a, b) {
+      return a.startDate > b.startDate ? a : b;
+    });
+
+  const programmeSpecialty = programme.curricula.find(
+    c => c.curriculumSubType === "MEDICAL_CURRICULUM"
```
Comment (DorisWongNHS) [inline src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts:19]: If there are more than 1 curricula with "MEDICAL_CURRICULUM" subtype, we may need to find the most recent as well

File: src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
   traineeProfile: TraineeProfile | null
 ): FormRPartA | null {
   if (!traineeProfile) {
     return null;
   }
 
   const pd = traineeProfile.personalDetails;
-  const programme = traineeProfile.programmeMemberships[0];
+  const programme = traineeProfile.programmeMemberships
+    .filter(pm => new Date(pm.startDate) < new Date())
```
Comment (hihilary) [inline src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts:13]: Just remembered Ashley told us the programmeMemberships could be future ones.

File: src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts (hunk 1 above)
Comment (hihilary) [reply src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts:19]: I agree with Doris, may need to apply this filter first.

File: src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts (hunk 2 above)
Comment (DorisWongNHS) [reply src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts:13]: Ya, for my understanding, the programmeMemberships can be current or future. Just compare to find the latest startDate should be alright. This line may not be needed.

File: src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts [hunk 3]
```diff
@@ -30,15 +39,15 @@ export function mapProfileToFormRPartAInitialValues(
...
     email: pd.email || "",
     isLeadingToCct: false,
-    programmeSpecialty: programme.curricula[0].curriculumName,
-    cctSpecialty1: "",
+    programmeSpecialty: programmeSpecialty?.curriculumName || "",
+    cctSpecialty1: programmeSpecialty?.curriculumName || "",
     cctSpecialty2: "",
     college: "",
-    completionDate: programme.endDate || "",
+    completionDate: programmeSpecialty?.curriculumCompletionDate || new Date(),
```
Comment (DorisWongNHS) [inline src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts:46]: If the trainee does not have a curriculum with "MEDICAL_CURRICULUM" subtype, the curriculumCompletionDate will also be filtered out?

File: src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts (hunk 1 above)
Comment (babul3012) [reply src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts:19]: @DorisWongNHS  / @hihilary  how do we get the most recent curriculum?

File: src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts (hunk 3 above)
Comment (babul3012) [reply src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts:46]: @DorisWongNHS  By filtered out do you mean that we should not populate the date?

File: src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts (hunk 2 above)
Comment (DorisWongNHS) [reply src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts:13]: If I did not misunderstand Ashley, I think the programmes can be 'not yet started'. That is Yafang's meaning of 'future programme'.

Comment (Judge40) [reply src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts:13]: We're basically look for programmes with an end date later than the current date, yes?

File: src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts (hunk 3 above)
Comment (babul3012) [reply src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts:46]: We do not populate if no subtype matches the condition.

File: src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts (hunk 2 above)
Comment (babul3012) [reply src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts:13]: @hihilary & @DorisWongNHS Ash confirmed that we have to consider future programmes as well(Applies the same as in curriculum as well). I will update the conditions accordingly.

File: src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts (hunk 1 above)
Comment (babul3012) [reply src/components/forms/formr-part-a/ProfileToFormRPartAInitialValuesMapping.ts:19]: we might have to add 'curriculumStartDate' property for curriculum.

**Pull request merged**